packages/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -713,16 +720,26 @@ export const UnifiedDataTable = ({
     [usedSelectedDocs, isFilterActive, rows, externalAdditionalControls]
   );
 
-  const showDisplaySelector = useMemo(
-    () =>
-      !!onUpdateRowHeight
-        ? {
-            allowDensity: false,
-            allowRowHeight: true,
-          }
-        : undefined,
-    [onUpdateRowHeight]
-  );
+  const showDisplaySelector = useMemo(() => {
+    const options: EuiDataGridToolBarVisibilityDisplaySelectorOptions = {};
+
+    if (onUpdateRowHeight) {
+      options.allowDensity = false;
+      options.allowRowHeight = true;
+    }
+
+    if (onUpdateSampleSize) {
+      options.allowResetButton = false;
```

Comment:
@davismcphee Since we have issues with the button https://github.com/elastic/kibana/issues/131130 and now we introduce additional settings which are not handled by eui code, we better hide the button. Originally suggested by @kertal

Reply:
yes, it did not work as expected, and introducing new settings, wouldn't make it better, so I suggested to remove it (not having a good way to handle it, at least back in the days)

Reply:
Ok, makes sense to remove it for now if it wasn't working as intended. Might be worth figuring out if we can fix it and add it back in later though. It would be convenient to be able to reset the sample size back to the saved search default after changing it on a dashboard and saving, which persists the updated value with the dashboard.